src/nginx_config.py

```diff
@@ -276,23 +189,29 @@ def _listen_args(self, port: int, ipv6: bool, ssl: bool, http2: bool) -> List[st
             args.append("http2")
         return args
 
-    def _servers(self, addresses_by_role: Dict[str, Set[str]]) -> List[Dict[str, Any]]:
+    def _servers(
+        self, addresses_by_role: Dict[str, Set[str]], tls: bool = False
+    ) -> List[Dict[str, Any]]:
         servers = []
         roles = addresses_by_role.keys()
 
-        if TempoRole.distributor.value in roles or TempoRole.all.value in roles:
+        if TempoRole.distributor.value in roles:
             for protocol, port in Tempo.receiver_ports.items():
-                servers.append(self._server(port, protocol.replace("_", "-"), "grpc" in protocol))
-        if TempoRole.query_frontend.value in roles or TempoRole.all.value in roles:
+                servers.append(
+                    self._server(port, protocol.replace("_", "-"), "grpc" in protocol, tls)
```

Comment:
instead of hardcoding the `'grpc' in protocol` thing, IMHO we should write a `_is_protocol_grpc -> bool` function where we explain why we're using this hack to determine if a protocol is grpc based. Also didn't we have already something like this in the tracing lib, to add the same information in the databag?